rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -49,31 +50,41 @@
   private DirectBufferPool bufferPool;
   private int directBufferSize;
   private int corruptedFileCount;
+  private int generationNumber;
```

Comment:
We dont need to keep another copy of `generationNumber`. Remove this and make `ReadRequestChain.generationNumber` scope `protected`